dataset/include/scipp/dataset/map_view.h

```diff
@@ -252,6 +252,10 @@ template <class Base, class Access> class MutableView : public Base {
 
   template <class VarOrView>
   void set(const typename Base::key_type key, VarOrView var) const {
+    if (!m_access.writeable() && Base::contains(key) &&
```

Comment:
Why do you check for "not writable" here?

Reply:
I didn't originally have it, but causes problems in a few places such as here:

positions end up unset on monitors https://github.com/scipp/scipp/blob/master/python/src/scipp/compat/mantid.py#L507

Reply:
Not sure I understand, can you elaborate.

Reply:
@OwenArnold Looking at this again after our call: It feels wrong to add a special case where calling `set` on a const view passes. Should we add loop-holes (even irrelant loop-holes) to our C++ library due to the way Python works (calling `__setitem__` after `__getitem__`)? I feel the proper way should be to fix this in the Python bindings, skipping the `__setitem__` for self-assignment. On the other hand, this is also weird since it kind of duplicates the self-assignment check we have in `Variable`? Not sure what I am concluding here.

Reply:
Yes. Agree feels wrong to fix a python issue at the c++ level. I'm not sure how we can reasonably do this at the point of `__setitem__` How can we overload correctly? One idea I had is to check that the data being assigned is the same for the indices over which it is being assigned maybe?

Reply:
Well, you just need to move this check to `__setitem__`, I think?

Reply:
Ok. I'll see how that pans out.

Reply:
Something like:
```python
if (self.contains(key))
  self[key].assign(var);
else
  self.set(key, var);
```
maybe?